lighthouse-cli/test/smokehouse/test-definitions/dobetterweb/dbw-expectations.js

```diff
@@ -11,6 +11,9 @@
  */
 const expectations = [
   {
+    networkRequests: {
+      length: 59,
```

Comment:
Does this seem high? The DT panel only shows 25 requests made during the page load.

Reply:
The higher number is coming from the three passes, but it does seem kind of high. From 

`defaultPass`: Lighthouse shows 26 network requests
 - 2 are `blob` and `filesystem` (shouldn't be hitting static-server)
 - 1 is `ajax.googleapis.com` for jquery (shouldn't be hitting static-server)
 - 23 are for `http://localhost:10200/`

(the now poorly named) `offlinePass`: Lighthouse shows 25 network requests
- `blob`, `filesystem`, and `ajax.googleapis.com` again
- 22 are for `http://localhost:10200/` (1 less because favicon isn't requested on this load)

`redirectPass`: Lighthouse shows 25 network requests
- `blob` and `ajax.googleapis.com` again (but filesystem is blocked by `blockedUrlPatterns` because it's a png)
- 23 are for `http://localhost:10200/`, but only 6 made it through to the server

That's only 51 by my count though :/

<details>

```
defaultPass: [
  {url: "http://localhost:10200/favicon.ico", statusCode: 404},
  {url: "http://localhost:10200/dobetterweb/unknown404.css?delay=200", statusCode: 404},
  {url: "http://localhost:10200/dobetterweb/third_party/aggressive-promise-polyfill.js", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-rotating.gif", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr3", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr2", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr1", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?iar2", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?iar1", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/fcp-delayer.js?delay=5000", statusCode: 404},
  {url: "http://localhost:10200/dobetterweb/empty.css", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/empty_module.js?delay=500", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.js", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.html", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.html", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?scriptActivated&delay=200", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=3000&capped", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=3000&async=true", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=2200", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=2000&async=true", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=100", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_disabled.css?delay=200&isdisabled", statusCode: 200},
  {url: "http://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js", statusCode: 200},
  {url: "filesystem:http://localhost:10200/temporary/empty-0.6220707096123763.png", statusCode: 200},
  {url: "blob:http://localhost:10200/a932e885-e346-4e8a-9e56-c3e37c4c427e", statusCode: 200
  }
],

offlinePass: [
  {url: "http://localhost:10200/dobetterweb/unknown404.css?delay=200", statusCode: 404},
  {url: "http://localhost:10200/dobetterweb/third_party/aggressive-promise-polyfill.js", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-rotating.gif", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr3", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr2", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr1", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?iar2", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?iar1", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/fcp-delayer.js?delay=5000", statusCode: 404},
  {url: "http://localhost:10200/dobetterweb/empty.css", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/empty_module.js?delay=500", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.js", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.html", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.html", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?scriptActivated&delay=200", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=3000&capped", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=3000&async=true", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=2200", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=2000&async=true", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=100", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_disabled.css?delay=200&isdisabled", statusCode: 200},
  {url: "http://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js", statusCode: 200},
  {url: "http://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js", statusCode: 200},
  {url: "filesystem:http://localhost:10200/temporary/empty-0.6121657698517076.png", statusCode: 200},
  {url: "blob:http://localhost:10200/fa7c68da-ce56-48ea-aaec-31b86bbf2c17", statusCode: 200
  }
],

redirectPass: [
  {url: "http://localhost:10200/dobetterweb/unknown404.css?delay=200", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/third_party/aggressive-promise-polyfill.js", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/lighthouse-rotating.gif", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr3", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr2", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?isr1", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?iar2", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg?iar1", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/lighthouse-480x318.jpg", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/fcp-delayer.js?delay=5000", statusCode: 404},
  {url: "http://localhost:10200/dobetterweb/empty.css", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/empty_module.js?delay=500", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.js", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.html", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.html", statusCode: 200},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?scriptActivated&delay=200", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=3000&capped", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=3000&capped", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=3000&capped", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=2200", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=2000&async=true", statusCode: -1},
  {url: "http://localhost:10200/dobetterweb/dbw_tester.css?delay=100", statusCode: -1},
  {url: "http://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js", statusCode: 200},
  {url: "blob:http://localhost:10200/b653a999-daa6-4573-b7f1-e97e2802ab63", statusCode: 200
  }
]
```

<details>

Reply:
ok, comparing to what `static-server` gives us:

<details>

```
[
  '/dobetterweb/dbw_tester.html',
  '/dobetterweb/dbw_tester.css?delay=2000&async=true',
  '/dobetterweb/dbw_tester.css?delay=100',
  '/dobetterweb/unknown404.css?delay=200',
  '/dobetterweb/dbw_tester.css?delay=2200',
  '/dobetterweb/dbw_disabled.css?delay=200&isdisabled',
  '/dobetterweb/dbw_tester.css?delay=3000&capped',
  '/dobetterweb/dbw_tester.css?delay=3000&async=true',
  '/dobetterweb/dbw_tester.js',
  '/dobetterweb/empty_module.js?delay=500',
  '/dobetterweb/fcp-delayer.js?delay=5000',
  '/dobetterweb/clock.appcache',
  '/dobetterweb/third_party/aggressive-promise-polyfill.js',
  '/dobetterweb/lighthouse-480x318.jpg?iar1',
  '/dobetterweb/lighthouse-480x318.jpg?iar2',
  '/dobetterweb/lighthouse-480x318.jpg?isr1',
  '/dobetterweb/lighthouse-480x318.jpg?isr2',
  '/dobetterweb/lighthouse-480x318.jpg?isr3',
  '/dobetterweb/lighthouse-480x318.jpg',
  '/dobetterweb/lighthouse-rotating.gif',
  '/dobetterweb/empty.css',
  '/dobetterweb/dbw_tester.css?scriptActivated&delay=200',
  '/dobetterweb/dbw_tester.html',
  '/dobetterweb/dbw_tester.css?delay=100',
  '/dobetterweb/unknown404.css?delay=200',
  '/dobetterweb/dbw_tester.css?delay=2200',
  '/dobetterweb/dbw_tester.css?delay=3000&capped',
  '/dobetterweb/dbw_tester.css?delay=2000&async=true',
  '/dobetterweb/dbw_tester.css?scriptActivated&delay=200',
  '/dobetterweb/dbw_tester.html',
  '/dobetterweb/dbw_tester.css?delay=2000&async=true',
  '/dobetterweb/dbw_tester.css?delay=100',
  '/dobetterweb/unknown404.css?delay=200',
  '/dobetterweb/dbw_tester.css?delay=2200',
  '/dobetterweb/dbw_disabled.css?delay=200&isdisabled',
  '/dobetterweb/dbw_tester.css?delay=3000&capped',
  '/dobetterweb/dbw_tester.css?delay=3000&async=true',
  '/dobetterweb/dbw_tester.js',
  '/dobetterweb/empty_module.js?delay=500',
  '/dobetterweb/fcp-delayer.js?delay=5000',
  '/dobetterweb/clock.appcache',
  '/dobetterweb/third_party/aggressive-promise-polyfill.js',
  '/dobetterweb/lighthouse-480x318.jpg?iar1',
  '/dobetterweb/lighthouse-480x318.jpg?iar2',
  '/dobetterweb/lighthouse-480x318.jpg?isr1',
  '/dobetterweb/lighthouse-480x318.jpg?isr2',
  '/dobetterweb/lighthouse-480x318.jpg?isr3',
  '/dobetterweb/lighthouse-480x318.jpg',
  '/dobetterweb/lighthouse-rotating.gif',
  '/dobetterweb/empty.css',
  '/dobetterweb/dbw_tester.css?scriptActivated&delay=200',
  '/dobetterweb/dbw_tester.html',
  '/dobetterweb/dbw_tester.html',
  '/dobetterweb/dbw_tester.js',
  '/dobetterweb/empty_module.js?delay=500',
  '/dobetterweb/fcp-delayer.js?delay=5000',
  '/dobetterweb/clock.appcache',
  '/dobetterweb/third_party/aggressive-promise-polyfill.js',
  '/dobetterweb/dbw_tester.html'
]
```

</details>

favicon apparently doesn't count for anything, but `/dobetterweb/clock.appcache` does (for all three passes). I forgot that was in that page, kind of annoying that it doesn't show up in the network requests (but it'll be removed from Chrome soon enough, so whatever).

That's 53, and the remaining 6 requests are re-requests of all the css files:

```
  '/dobetterweb/dbw_tester.css?delay=100',
  '/dobetterweb/unknown404.css?delay=200',
  '/dobetterweb/dbw_tester.css?delay=2200',
  '/dobetterweb/dbw_tester.css?delay=3000&capped',
  '/dobetterweb/dbw_tester.css?delay=2000&async=true',
  '/dobetterweb/dbw_tester.css?scriptActivated&delay=200',
```

They appear to be happening in `defaultPass`, but I verified there's no `Network.requestWillBeSent` events in any of the passes that would account for these requests, so either they're happening outside of `pass()` (so not being picked up by `NetworkRecorder`) or they're some out-of-band request (like the appcache manifest is) that DevTools doesn't get info on via `Network` events

Reply:
OK, can confirm that those six requests are happening at the end of `defaultPass` (or at least not during the other passes...deleted the other passes and those six requests were still recorded by `static-server`).

It's possible this is one of the bugs this assertion was being added to prevent :) I'll keep looking but if anyone else has any ideas as to the source of these, please let us know.

Reply:
I would expect it to be 50 for `networkRequests` because we ignore the favicon. These are the 56 requests made according to smoke:
```
"/dobetterweb/dbw_tester.html",
"/dobetterweb/dbw_tester.css?delay=2000&async=true",
"/dobetterweb/dbw_tester.css?delay=100",
"/dobetterweb/unknown404.css?delay=200",
"/dobetterweb/dbw_tester.css?delay=2200",
"/dobetterweb/dbw_disabled.css?delay=200&isdisabled",
"/dobetterweb/dbw_tester.css?delay=3000&capped",
"/dobetterweb/dbw_tester.css?delay=3000&async=true",
"/dobetterweb/dbw_tester.js",
"/dobetterweb/empty_module.js?delay=500",
"/dobetterweb/fcp-delayer.js?delay=5000",
"/dobetterweb/clock.appcache",
"/dobetterweb/third_party/aggressive-promise-polyfill.js",
"/dobetterweb/lighthouse-480x318.jpg?iar1",
"/dobetterweb/lighthouse-480x318.jpg?iar2",
"/dobetterweb/lighthouse-480x318.jpg?isr1",
"/dobetterweb/lighthouse-480x318.jpg?isr2",
"/dobetterweb/lighthouse-480x318.jpg?isr3",
"/dobetterweb/lighthouse-480x318.jpg",
"/dobetterweb/lighthouse-rotating.gif",
"/dobetterweb/empty.css",
"/dobetterweb/dbw_tester.css?scriptActivated&delay=200",
"/dobetterweb/dbw_tester.html",
"/dobetterweb/dbw_tester.css?delay=100",
"/dobetterweb/unknown404.css?delay=200",
"/dobetterweb/dbw_tester.css?delay=2200",
"/dobetterweb/dbw_tester.css?delay=3000&capped",
"/dobetterweb/dbw_tester.css?delay=2000&async=true",
"/dobetterweb/dbw_tester.css?scriptActivated&delay=200",
"/dobetterweb/dbw_tester.html",
"/dobetterweb/dbw_tester.css?delay=2000&async=true",
"/dobetterweb/dbw_tester.css?delay=100",
"/dobetterweb/unknown404.css?delay=200",
"/dobetterweb/dbw_tester.css?delay=2200",
"/dobetterweb/dbw_disabled.css?delay=200&isdisabled",
"/dobetterweb/dbw_tester.css?delay=3000&capped",
"/dobetterweb/dbw_tester.css?delay=3000&async=true",
"/dobetterweb/dbw_tester.js",
"/dobetterweb/empty_module.js?delay=500",
"/dobetterweb/fcp-delayer.js?delay=5000",
"/dobetterweb/clock.appcache",
"/dobetterweb/third_party/aggressive-promise-polyfill.js",
"/dobetterweb/lighthouse-480x318.jpg?iar1",
"/dobetterweb/lighthouse-480x318.jpg?iar2",
"/dobetterweb/lighthouse-480x318.jpg?isr1",
"/dobetterweb/lighthouse-480x318.jpg?isr2",
"/dobetterweb/lighthouse-480x318.jpg?isr3",
"/dobetterweb/lighthouse-480x318.jpg",
"/dobetterweb/lighthouse-rotating.gif",
"/dobetterweb/empty.css",
"/dobetterweb/dbw_tester.css?scriptActivated&delay=200",
"/dobetterweb/dbw_tester.html",
"/dobetterweb/dbw_tester.html",
"/dobetterweb/dbw_tester.js",
"/dobetterweb/empty_module.js?delay=500",
"/dobetterweb/fcp-delayer.js?delay=5000",
"/dobetterweb/clock.appcache",
"/dobetterweb/third_party/aggressive-promise-polyfill.js",
"/dobetterweb/dbw_tester.html"
```

Reply:
Unfortunately this isn't a solution or an answer to the mystery. It looks like a similar problem is occurring on `http://localhost:10200/preload.html`. The requests are:
```
"/preload.html",
"/perf/level-2.js?warning&delay=500",
"/perf/preload_style.css",
"/perf/preload_tester.js",
"/perf/lobster-v20-latin-regular.woff2",
"/perf/level-2.js?delay=500",
"/perf/level-3.js",
"/perf/level-2.js?warning&delay=500",
"/perf/preload_style.css"
```

`/perf/preload_style.css` is fetched twice.

Reply:
hmm, maybe a gatherer?

Reply:
simply opening devtools on the page makes the extra requests to CSS files.

![image](https://user-images.githubusercontent.com/4071474/114788279-69977980-9d3e-11eb-8fde-c6b6c7de7c38.png)

Reply:
Looks like it's `CSS.enable` in particular. If you don't run the gatherers `CSSUsage`, `ImageElements`, and `FontSize` you don't get those extra requests.

(I assume opening DevTools calls `CSS.enable` by default)

Reply:
To close this up, these extra requests are intentional. Chrome is aggressive with evicting raw stylesheet contents from memory, so DevTools reacquires them when `CSS.enable` is called. It hits the disk cache first (original change: https://codereview.chromium.org/24151005), but we don't set cache headers on our requests, so the test stylesheets aren't in the disk cache and are re-fetched.

A comment is probably sufficient to document this, but I guess we'll see how annoying this number is to maintain in the future :) Does the next person touching `dbw_tester` have to go through the same logging-URL exercise to make sure the new number is reasonable?